### PR TITLE
Add workflow, link checker and modify Readme

### PR DESCRIPTION
**Pull request by Divyansh42**
### Changes

- Add test workflow to test image build from Dockerfile.
- Add Link checker to check for dangling links in markdown
- Modify Readme to divide inputs for build with and without dockerfile

### CODE REVIEW

File: .github/workflows/dockerfile_build.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# This workflow will perform a test whenever there
+# is some change in code done to ensure that the changes
+# are not buggy and we are getting the desired output.
+name: Test Build with dockerfile
+on: [push, pull_request, workflow_dispatch]
+env:
+  PROJECT_DIR: spring-petclinic
```
Comment (tetchel) [inline .github/workflows/dockerfile_build.yml:7]: This is not used

File: .github/workflows/dockerfile_build.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  IMAGE_NAME: spring-petclinic
```
Comment (tetchel) [inline .github/workflows/dockerfile_build.yml:8]: Update this to hello-world or something

File: .github/workflows/dockerfile_build.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  MVN_REPO_DIR: ~/.m2/repository
```
Comment (tetchel) [inline .github/workflows/dockerfile_build.yml:9]: Also not used

File: .github/workflows/dockerfile_build.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+jobs:
+  build:
+    name: Build image using Buildah
+    runs-on: ubuntu-latest
```
Comment (tetchel) [inline .github/workflows/dockerfile_build.yml:14]: Can we just use 20.04 because the warning about `latest` changing is annoying

File: README.md [hunk 5]
```diff
@@ -1,6 +1,8 @@
 # buildah-build
 [![CI checks](https://github.com/redhat-actions/buildah-build/workflows/CI%20checks/badge.svg)](https://github.com/redhat-actions/buildah-build/actions?query=workflow%3A%22CI+checks%22)
-[![Verify Build](https://github.com/redhat-actions/buildah-build/workflows/Test%20Build/badge.svg)](https://github.com/redhat-actions/buildah-build/actions?query=workflow%3A%22Test+Build%22)
+[![Test Build with dockerfile](https://github.com/redhat-actions/buildah-build/workflows/Test%20Build%20with%20dockerfile/badge.svg)](https://github.com/redhat-actions/buildah-build/actions?query=workflow%3A%22Test+Build+with+dockerfile%22)
```
Comment (tetchel) [inline README.md:3]: Can you maybe shorten the workflow names. They will look too long on these badges

File: README.md [hunk 6]
```diff
@@ -15,108 +17,31 @@ After building your image, use [push-to-registry](https://github.com/redhat-acti
...
-envs: |
-  GOPATH=/root/buildah/go</pre>
-    </td>
-  </tr>
-</table>
+### Inputs for build from dockerfile
+
+| Input Name | Description | Default |
+| ---------- | ----------- | ------- |
+| build-args | Build arguments to pass to the Docker build using `--build-arg`, if using a Dockerfile that requires ARGs. Uses the form `arg_name=arg_value`, and separate arguments with newlines. | None
```
Comment (tetchel) [inline README.md:24]: Uses the form -> Use the form

File: README.md [hunk 7]
```diff
@@ -15,108 +17,31 @@ After building your image, use [push-to-registry](https://github.com/redhat-acti
...
+| context | Path to directory to use as the build context. | `.`
+| dockerfiles | The list of Dockerfile paths to perform a build using docker instructions. This is a multiline input to allow multiple Dockerfiles. | **Must be provided**
+| image | Name to give to the output image. | **Must be provided**
+| oci | Build the image using the OCI format, instead of the Docker format. By default, this is `false`, because images built using the OCI format have issues when published to `Dockerhub`. | `false`
```
Comment (tetchel) [inline README.md:28]: Don't code-ify Dockerhub

File: README.md [hunk 8]
```diff
@@ -15,108 +17,31 @@ After building your image, use [push-to-registry](https://github.com/redhat-acti
...
+| tags | The tags of the image to build. For multiple tags, separate by a space. For example, `latest ${{ github.sha }}` | `latest`
+
+### Inputs for build without dockerfile
+
+| Input Name | Description | Default |
+| ---------- | ----------- | ------- |
+| base-image | The base image to use for the container. | **Must be provided**
+| content | The content to copy inside the container to create the final image. This is a multiline input to allow you to copy more than one file/directory.| No
```
Comment (tetchel) [inline README.md:36]: No default -> "None"

Comment (tetchel) [inline README.md:36]: "Paths to files or directories to copy inside the container to create the file image. This is a multiline input to allow you to copy multiple files/directories."

File: README.md [hunk 9]
```diff
@@ -15,108 +17,31 @@ After building your image, use [push-to-registry](https://github.com/redhat-acti
...
+| context | Path to directory to use as the build context. | `.`
+| entrypoint | The entry point to set for the container. This is a multiline input; split arguments across lines. | None
+| envs | The environment variables to be set when running the container. This is a multiline input to add multiple environment variables. | None
+| image | Name to give to the output image. | **Must be provided**
+| oci | Build the image using the OCI format, instead of the Docker format. By default, this is `false`, because images built using the OCI format have issues when published to `Dockerhub`. | `false`
```
Comment (tetchel) [inline README.md:41]: don't code-ify dockerhub

File: README.md [hunk 10]
```diff
@@ -216,8 +141,8 @@ This is an open source project open to anyone. This project welcomes contributio
 
 ## Feedback & Questions
 
-If you discover an issue please file a bug in [GitHub issues](https://github.com/redhat-actions/buildah/issues) and we will fix it as soon as possible.
+If you discover an issue please file a bug in [GitHub issues](https://github.com/redhat-actions/buildah-build/issues) and we will fix it as soon as possible.
 
 ## License
 
-MIT, See [LICENSE](https://github.com/redhat-actions/buildah/blob/main/LICENSE.md) for more information.
+MIT, See [LICENSE](https://github.com/redhat-actions/buildah-build/blob/main/LICENSE) for more information.
```
Comment (tetchel) [inline README.md:148]: Just use ./LICENSE

File: README.md [hunk 11]
```diff
@@ -216,8 +141,8 @@ This is an open source project open to anyone. This project welcomes contributio
 
 ## Feedback & Questions
 
-If you discover an issue please file a bug in [GitHub issues](https://github.com/redhat-actions/buildah/issues) and we will fix it as soon as possible.
+If you discover an issue please file a bug in [GitHub issues](https://github.com/redhat-actions/buildah-build/issues) and we will fix it as soon as possible.
```
Comment (tetchel) [inline README.md:144]: Honestly this section and the License section could just be removed.

Comment (tetchel) [reply README.md:144]: And the Contributing section.

**Review by tetchel — CHANGES_REQUESTED** [review]

**Comment by Divyansh42** [comment]
Oops.. it seems I did something wrong. I remember I have changes most of the things that there in reviews :sweat_smile:

File: README.md (hunk 5 above)
Comment (Divyansh42) [reply README.md:3]: What about `Dockerfile build` and `Scratch build` :thinking: 
or may me `Build from Scratch` & `Build from Dockerfile`
I know second option is not that short.

File: README.md (hunk 11 above)
Comment (Divyansh42) [reply README.md:144]: We can, as this is well known thing in open source projects.

File: README.md (hunk 5 above)
Comment (tetchel) [reply README.md:3]: how about `buildah` and `buildah bud`

Comment (Divyansh42) [reply README.md:3]: I don't think this clarifies the thing that we are achieving by this workflow.

Comment (tetchel) [reply README.md:3]: "Build" and "Build from Dockerfile" ?

Comment (Divyansh42) [reply README.md:3]: This is good :+1:

**Review by tetchel — APPROVED** [review]

**Pull request merged**